src/generalSettings/index.ts

```diff
@@ -13,10 +16,10 @@ export class GeneralSettingsEndpoint extends BaseApi {
     return responseHandlers.list(response);
   }
 
-  public readRaw(requestParams?: RequestParams<GeneralSetting> | Object): Promise<ApiResponseOnSuccess<GeneralSetting[]>> {
-    const params = requestParams instanceof RequestParams ? requestParams.getParams() : requestParams;
+  public async readRaw(requestParams: RequestParams<GeneralSetting>): Promise<ReadRawResponse<GeneralSetting>> {
+    const params = requestParams.getParams();
     const response = this._get<GeneralSetting[]>(`${this.getResourceName()}/read`, { params });
-    return responseHandlers.toApiResponse(response);
+    return createReadRawResponse<GeneralSetting>(createResourceResponse(await resolveAxiosResponse(response)), requestParams);
```

Comment:
Note: these 3 functions will be merged later into one. When we now they are stable